### DISCUSSION TITLE
How to enable the auto update without user notification

### DISCUSSION THREAD

**Discussion by mani2care** [rxhanson/Rectangle#1443] [General]
How to enable the auto-update without user notification is there any plist or json file for the enabling the auto enable and install the  latest version ?  via JAMF ?

**Comment by rxhanson** [comment]
Rectangle uses Sparkle to handle the updates. 

I think you just need to set this value to true, but have never thoroughly tested this myself.

```
defaults write com.knollsoft.Rectangle SUAutomaticallyUpdate -bool true
```

Here is the relevant doc: https://sparkle-project.org/documentation/customization/

**Reply by mani2care** [reply]
```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
  <dict>
    <key>SUEnableAutomaticChecks</key>
    <true/>
    <key>SUAutomaticallyUpdate</key>
    <true/>
    <key>SUAllowsAutomaticUpdates</key>
    <true/>
     <key>SUScheduledCheckInterval</key>
     <integer>86400</integer>
  </dict>
</plist>
```

i have set this as per the document but still im not seeing the any changes app is not updating it self did mistaken any thing or missing any ?

**Reply by rxhanson** [reply]
Perhaps the proper plist isn't getting written to? It needs to be the one that has all of the other Rectangle configuration in it. Maybe try:

```
defaults write com.knollsoft.Rectangle.plist SUAutomaticallyUpdate -bool true
```

**Comment by mani2care** [comment]
It’s doing only enable auto update and autoinstall not happening while open
the app its just asking user to update the app there is no enforcement key ?

Thanks & Regards,

Manikandan R

**Comment by rxhanson** [comment]
Also, I'm not familiar with JAMF, but it's likely that any external tool would be leveraging the existing Sparkle functionality to be aware of updates. If you're not getting what you want out of the Sparkle options, then my other suggestion would be to install the app via brew and from there you could script update checking and installation via brew. I don't have specific examples of how to do this, but I suspect it should be pretty straightforward to figure out.

**Comment by mani2care** [comment]
Thanks for the tip!

However I’m looking from application end functionality and documentation
for the app updates
